scripts/main/init.js

```diff
@@ -151,6 +151,35 @@ $(document).ready(function() {
 
 	})
 
+	// Paste upload
+	.on('paste', function (e) {
+		if (e.originalEvent.clipboardData.items) {
+			if (!album.isUploadable()) {
+				return false;
+			}
+
+			// Close open overlays or views which are correlating with the upload
+			if (visible.photo()) lychee.goto(album.getID());
+			if (visible.contextMenu()) contextMenu.close();
+
+			if (e.originalEvent.clipboardData.items.length > 0) {
+				const items = e.originalEvent.clipboardData.items;
+				let imagesToUpload = [];
+
+				// Search clipboard items for an image
+				for (let i = 0; i < items.length; i++) {
```

Comment:
Are you a C programmer by training? :smiley:

`forEach`/`$.each` would be a more JavaScript-y way of iterating over these...

Reply:
Oh I'm just used to using a for loop. I always have this "It's faster that way" in my mind.
Will change.

Reply:
Okay nevermind it doesn't work with `forEach`/`$.each` because `items` isn't an array but a `DataTransferItemList`

Reply:
My bad...